dart/common/Uri.h

```diff
@@ -97,16 +100,34 @@ class Uri final
   /// Fragment, e.g. the part of the URI after the #
   UriComponent mFragment;
 
+  /// Constructor
+  Uri() = default;
+
+  /// Constructor that takes a URI string. Internally, this is equivalent to
+  /// calling fromString(_input) after default constructor.
+  Uri(const std::string& _input);
```

Comment:
Also, I suppose backwards compatibility would require the constructor to call ``fromPath`` instead of ``fromString``. I guess this was the original motivation for having ``fromPathOrString``.